### DISCUSSION TITLE
ADAL Deprication for Microsoft Login Control

### DISCUSSION THREAD

**Discussion by hbriggs003** [MscrmTools/XrmToolBox#1333] [General]
Is there a plan to update the application for Microsoft Login Control based on MSAL rather than ADAL? ADAL is deprecated and will likely be unavailable beginning 6/24.

**Comment by MscrmTools** [comment]
I wasn't aware of this change but if it's "only" deprecated and not removed, then we have some time to update XrmToolBox to the latest connection controls

**Reply by hbriggs003** [reply]
The expected timeline for removal is June 2024, from what I have been told.

**Reply by hbriggs003** [reply]
I will see if I can obtain any additional information regarding the date.

**Comment by MattB-msft** [comment]
@MscrmTools  
You are correct, it is deprecated and has not been removed. 

Insofar as switching over to use the Dataverse ServiceClient / MSAL there are several limitations: 
Dataverse ServiceClient supports AD and JWT auth only.  
It does not support (or have the code for) WS-TRUST base authentication, which we refer to IFD with Claims.  We have re

You will find that support for ADFS 2016 (so customers using Dynamics CRM OnPrem + Windows 2016 or older ADFS) is missing.  ADAL.net must be used. 

if you're willing to take XrmToolbox to an Online | AD only feature set, then you could defiantly update to DVSC + MSAL
We are already doing that for PAC CLI, dev tools and such, and will soon be shipping updates to the plugin registration tool that will switch over to DVSC.

**Reply by BetimBeja** [reply]
@MattB-msft what does `Plugin Registration Tool` use?
I believe that the moment XrmToolBox becomes cloud only, then on-Premises CRM is doomed 😅

**Reply by MattB-msft** [reply]
Plugin Registration Tool currently uses CrmServiceClient (ADAL.net) though very shortly we will be forking it from Microsoft.CrmSdk.PluginRegistrationTool to Microsoft.PowerPlatform.Dataverse.PluginRegistrationTool  and moving it to Dataverse ServiceClient ( MSAL ) 

The old package will be marked as "deprecated" and guidance provided to use for OnPrem Only.

**Reply by MscrmTools** [reply]
@MattB-msft Do I understand it correctly if I say that I will have to migrate XrmToolBox to DVSC sometimes in the near future? 

If yes, that means that each and every XrmToolBox plugin will have to adapt to use DSVC instead of CrmServiceClient (as they are not compatible) so it can work.

So, that will imply a new XrmToolBox V2 with only tools that would have been updated to be used with DSVC and that would be a cloud only version.
